lib/utils.pm

```diff
@@ -1509,6 +1510,54 @@ sub script_retry {
     return $ret;
 }
 
+=head2 script_output_retry
+
+ script_output_retry($cmd, [proceed_on_failure => $proceed_on_failure], [retry => $retry], [delay => $delay], [timeout => $timeout], [die => $die]);
+
+Repeat command until expected result or timeout. Return the output of the command on success.
+
+C<$expect> refers to the expected command exit code and defaults to C<0>.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+The command must return within C<$timeout> seconds (default: 25).
+
+If the command doesn't return C<$expect> after C<$retry> retries,
+this function will die, if C<$die> is set.
+
+Example:
+
+ script_output_retry('ping -c1 -W1 machine', retry => 5);
+
+=cut
+sub script_output_retry {
+    my ($cmd, %args) = @_;
+    my $proceed_on_failure = $args{proceed_on_failure} // 0;
+    my $retry              = $args{retry}              // 10;
+    my $delay              = $args{delay}              // 30;
+    my $timeout            = $args{timeout}            // 30;
+    my $die                = $args{die}                // 1;
+
+    my $ret;
+    my $exec = "timeout " . ($timeout - 3) . " $cmd";
+    for (1 .. $retry) {
+        eval {
+            $ret = script_output($exec, timeout => $timeout, proceed_on_failure => $proceed_on_failure);
+            1;
+        }
+          or do {
+            die("Waiting for Godot: $cmd") if $retry == $_ && $die == 1;
```

Comment:
I think we should use `proceed_on_failure` here or?
And `proceed_on_failure` to the `script_output()` call should be always 1.
WDYT?

Reply:
A other hint:
```
for (1 .. $retry){
    my $ret = eval { script_output( ...) };
    if (!$ret) {
        sleep($delay);
    } else {
        return $ret;
    }
}
die ("Waiting for Godot") if $die;
```